factor/token/classes/factor.php

```diff
@@ -144,14 +144,19 @@ public function global_definition_after_data($mform) {
         // First thing, we need to decide on whether we should show the checkbox.
         $noproperty = !property_exists($SESSION, 'tool_mfa_factor_token');
         $nostate = $this->get_state() !== \tool_mfa\plugininfo\factor::STATE_PASS;
+        // We shouldn't show if the current factor does not give points.  Nothing to "trust" in that case.
+        $factor = \tool_mfa\plugininfo\factor::get_next_user_factor();
+        $weight = $factor->get_weight();
 
-        if ($noproperty && $nostate) {
+        if ($noproperty && $nostate && $weight !== 0) {
             $expiryconfig = is_siteadmin() ? 'adminexpiry' : 'expiry';
             $expiry = get_config('factor_token', $expiryconfig);
             $expirystring = format_time($expiry);
-            $mform->addElement('advcheckbox', 'factor_token_trust', '', get_string('form:trust', 'factor_token', $expirystring));
-            $mform->setType('factor_token_trust', PARAM_BOOL);
-            $mform->setDefault('factor_token_trust', true);
+            if ($expiry !== 0) {
```

Comment:
the expiry check should probably move to line 151 IMO to avoid double nesting

e.g. 
```
$expiryconfig = is_siteadmin() ? 'adminexpiry' : 'expiry';
$expiry = get_config('factor_token', $expiryconfig);

if ($noproperty && $nostate && $weight !== 0 && !empty($expiry)) {
... add form elements
}
```